frontend/src/component/common/PremiumFeature/PremiumFeature.tsx

```diff
@@ -0,0 +1,84 @@
+import { ReactComponent as ProPlanIcon } from 'assets/icons/pro-enterprise-feature-badge.svg';
+import { Box, Link, styled, Typography } from '@mui/material';
+import { usePlausibleTracker } from 'hooks/usePlausibleTracker';
+
+const PremiumFeatureWrapper = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    flexDirection: 'column',
+    padding: theme.spacing(1, 0.5),
+}));
+
+const StyledTitle = styled(Typography)(({ theme }) => ({
+    display: 'inline-flex',
+    alignItems: 'center',
+    fontWeight: theme.fontWeight.bold,
+    fontSize: theme.fontSizes.smallBody,
+    gap: theme.spacing(1),
+}));
+
+const StyledBody = styled(Typography)(({ theme }) => ({
+    fontSize: theme.fontSizes.smallBody,
+    margin: theme.spacing(1, 0),
+}));
+
+const StyledLink = styled(Link)(({ theme }) => ({
+    fontSize: theme.fontSizes.smallBody,
+    width: 'fit-content',
+}));
+
+export enum FeatureLevelTitle {
+    PRO = 'Pro & Enterprise feature',
+    ENTERPRISE = 'Enterprise feature',
+}
+
+export enum PlausibleOrigin {
+    PROJECT = 'Projects',
+    ACCESS = 'Access',
+    CHANGE_REQUEST = 'Change Request',
+}
+
+export interface PremiumFeatureProps {
+    children: React.ReactNode;
+    origin: PlausibleOrigin;
```

Comment:
From your code it seems like `origin` should be optional. If not set, we don't track that specific event.
If we want to make `origin` required and always track, then we should remove the `if (origin)` condition below.

```suggestion
    origin?: PlausibleOrigin;
```